docs/source/llms/chat-model-guide/index.rst

```diff
@@ -0,0 +1,802 @@
+
+                output = ChatResponse(
+                    choices=[ChatChoice(index=0, message=ChatMessage(**judge_response))],
+                    usage={},
+                    model=judge_params.get("endpoint", "unknown"),
+                )
+
+                root_span.set_outputs(output)
+
+            return output
+
+        def _get_model_params(self, role: str) -> dict:
+            """
+            Retrieves model parameters for a given role.
+
+            Args:
+                role (str): The role of the agent (e.g., "oracle" or "judge").
+
+            Returns:
+                dict: A dictionary of parameters for the agent.
+            """
+            role_config = self.models.get(role, {})
+
+            return {
+                "temperature": role_config.get("temperature", 0.5),
+                "max_tokens": role_config.get("max_tokens", 500),
+            }
+
+Now that we have our model defined, the process of logging it has only a single step that is required to be taken before logging: 
+we need to define the configuration for our model to be initialized with. This is done by defining our ``model_config`` configuration.
+
+Setting our ``model_config`` values
+-----------------------------------
+
+Before logging the model, we need to define the configuration that governs the behavior of our model's agents. This decoupling of configuration from the core logic of the model allows us to easily test and compare different agent behaviors without needing to modify the model implementation. By using a flexible configuration system, we can efficiently experiment with different settings, making it much easier to iterate and fine-tune our model.
+
+Why Decouple Configuration?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In the context of Generative AI (GenAI), agent behavior can vary greatly depending on the instruction sets and parameters (such as ``temperature`` or 
+``max_tokens``) given to each agent. If we hardcoded these configurations directly into our model's logic, each new test would require changing the 
+model's source code, leading to:
+
+- **Inefficiency**: Changing source code for each test slows down the experimentation process.
+- **Increased Risk of Errors**: Constantly modifying the source increases the chance of introducing bugs or unintended side effects.
+- **Lack of Reproducibility**: Without a clear separation between code and configuration, tracking and reproducing the exact configuration used for 
+  a particular result becomes challenging.
+
+By setting these values externally via the ``model_config`` parameter, we make the model flexible and adaptable to different test scenarios. 
+This approach also integrates seamlessly with MLflow's evaluation tools, such as :py:func:`mlflow.evaluate`, which allows you to compare model 
+outputs across different configurations systematically.
+
+Defining the Model Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The configuration consists of two main sections:
```

Comment:
I think we can also mention that the config can be defined in a yaml file, so users just need to specify path to the file when logging

Reply:
excellent idea. Added!